src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -686,7 +704,7 @@ private boolean checkExtends(Type a, Type bound) {
              return true;
          } else if (!a.hasTag(WILDCARD)) {
              a = types.cvarUpperBound(a);
-             return types.isSubtype(a, bound);
+             return types.isBoundedBy(a, bound, (t, s, w) -> types.isSubtype(t, s));
```

Comment:
This is a good change - I'm a bit scared about the other calls in this routine, to `isCastable` and `notSoftSubtype`. These type relations are mostly doing things outside the spec, and we probably need to sprinkle `isBounded` in there as well.